code/game/mecha/mecha_decal_kits.dm

```diff
@@ -0,0 +1,566 @@
+
+////////////////////
+// Decal Patterns //
+////////////////////
+// Container object which may hold more than one pattern. Must be installed in a paint gun prior to use.
+// Decals are added to the mech in sequential order.
+
+/obj/item/mecha_decal_container
+    name = "exosuit decal chip"
+    desc = "A wafer-thin data storage card made to slot into an exosuit paint gun."
+    icon = 'icons/mecha/mecha_decals.dmi'
+    icon_state = "pattern-chip"
+    w_class = WEIGHT_CLASS_TINY
+
+    var/decal_name                                      // The name shown by the decal UI.
+    var/list/datum/mecha/mecha_decal/decals             // Contained decals
+    var/list/obj/mecha/compatible_mecha = list()        // Mechs this pattern may be applied to.
+    var/basecoat                                        // If set to a colour, overwrites the mech's base colour on application.
+    var/glow                                            // If set to a colour, overwrites the mech's glow colour on application.
+    var/deletable = FALSE                               // If true, the paint gun UI will present an option to delete this item. Always true for custom decal kits.
+
+/obj/item/mecha_decal_container/proc/install_on_mecha(obj/mecha/M)
+    if(decals && (M.type in compatible_mecha))
+        for(var/datum/mecha/mecha_decal/decal in decals)
+            M.add_decal(decal.clone())
+        if(basecoat)
+            M.basecoat_colour = basecoat
+        if(glow)
+            M.glow_colour = glow
+        return TRUE
+    else
+        to_chat(usr, "<span class='warning'>[src] is not compatible with [M]!</span>")
+        return FALSE
+
+/obj/item/mecha_decal_container/examine()
+    . = ..()
+
+// PATTERN DEFINITIONS
+
+// Ripley Compatible
+/obj/item/mecha_decal_container/Initialize()
+    ..()
+    name = "exosuit decal chip ([decal_name])"
+
+/obj/item/mecha_decal_container/titan/Initialize()
+    decal_name = "Titan's Fist Ripley"
+    decals = list(new/datum/mecha/mecha_decal/titan, new/datum/mecha/mecha_decal/titaneyes)
+    compatible_mecha = COMPATIBLE_MECHA_RIPLEY
+    ..()
+
+/obj/item/mecha_decal_container/nanotrasen_logo/Initialize()
+    decal_name = "NT Logo"
+    decals = list(new/datum/mecha/mecha_decal/nanotrasen_logo)
+    compatible_mecha = COMPATIBLE_MECHA_COMMON
+    ..()
+
+/obj/item/mecha_decal_container/syndicate_logo/Initialize()
+    icon_state = "pattern-chip-syn"
+    decal_name = "Syndicate Logo"
+    desc = "A wafer-thin card made to slot into an exosuit paint gun. This one looks quite suspicious."
+    decals = list(new/datum/mecha/mecha_decal/syndicate_logo)
+    compatible_mecha = COMPATIBLE_MECHA_COMMON
+    ..()
+
+/obj/item/mecha_decal_container/stripes/Initialize()
+    decal_name = "Stripes"
+    decals = list(new/datum/mecha/mecha_decal/stripes)
+    compatible_mecha = COMPATIBLE_MECHA_RIPLEY
+    ..()
+
+/obj/item/mecha_decal_container/deathripley/Initialize()
+    decal_name = "Death Ripley"
```

Comment:
I'm not sure how I feel about setting base variables inside `Initialize` - it means that it won't grab the right details if you use the `initial(foo.decal_name)`, for example. Is there a reason you can't set these as variables on the specific path?

Reply:
It was laziness mostly. This should be fine now.